### PR TITLE
add overload in xref spec.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @ansyral @hellosnow @qinezh

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs [hunk 1]
```diff
@@ -244,6 +245,117 @@ private static XRefSpec GetXRefInfo(ItemViewModel item, string key)
...
+                    uidBody = uidBody.Remove(index);
+                }
+            }
+            uidBody = System.Text.RegularExpressions.Regex.Replace(uidBody, @"``\d+$", string.Empty);
+            var overload = new XRefSpec
+            {
+                Uid = uidBody + "*",
+                Href = key,
+                CommentId = "Overload:" + uidBody,
+            };
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs]: not used

File: src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs [hunk 2]
```diff
@@ -244,6 +245,117 @@ private static XRefSpec GetXRefInfo(ItemViewModel item, string key)
...
+                    result.Name = item.Name.Remove(index);
+                }
+                else
+                {
+                    result.Name = item.Name;
+                }
+            }
+            if (!string.IsNullOrEmpty(item.NameForCSharp))
+            {
+                var index = item.NameForCSharp.IndexOfAny(new char[] { '(', '[' });
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs]: >item.NameForCSharp.IndexOfAny(new char[] { '(', '[' } [](start = 28, length = 53)

is it necessary to extract one method to remove the index?

**Comment by hellosnow** [comment]
:shipit:

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs [hunk 3]
```diff
@@ -244,6 +245,117 @@ private static XRefSpec GetXRefInfo(ItemViewModel item, string key)
...
+            // todo : remove when overload is ready in yaml file.
+            if (item.Type != null)
+            {
+                switch (item.Type.Value)
+                {
+                    case MemberType.Property:
+                    case MemberType.Constructor:
+                    case MemberType.Method:
+                    case MemberType.Operator:
+                        yield return GenerateOverloadXrefSpec(item, key, result);
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs]: >result [](start = 73, length = 6)

update

File: src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs (hunk 2 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs]: We will remove this code later.

---
In reply to: [80480966](https://github.com/dotnet/docfx/pull/735#discussion_r80480966) [](ancestors = 80480966)

**Pull request merged**